diff --git a/libs/typedoc.ts b/libs/typedoc.ts
index 9030cc81c2ff022ad3e9e507d8397fd8a02e6034..e10d661b8c02fb177ff089b64ad11b31d90ca20d 100644
--- a/libs/typedoc.ts
+++ b/libs/typedoc.ts
@@ -140,6 +140,17 @@ function onRendererPageEnd(event: MarkdownPageEvent, pagination: boolean) {
     return true
   }
 
+  const old = event.url
+  const match = /([/\\])(?=(.))(?:functions|variables|interfaces|type-aliases|classes|enums)[/\\]([^.]+)\.md$/.exec(old)
+  const newURL = match ? old.slice(0, match.index) + match[1] + match[2] + "-" + match[3] + ".md" : old
+  const isFuncVar = match && (match[2] == "v" || match[2] == "f")
+
+  event.filename = event.filename.slice(0, -old.length) + newURL
+  event.url = newURL
+  event.contents = event.contents
+    ?.replace(/\/(?=(.))(?:functions|variables|interfaces|type-aliases|classes|enums)\//g, "/$1-")
+    .replace(/(?:\.\.\/)+/g, "/")
+
   if (!event.frontmatter) {
     event.contents = addFrontmatter(event.contents, {
       editUrl: false,
@@ -147,6 +158,7 @@ function onRendererPageEnd(event: MarkdownPageEvent, pagination: boolean) {
       prev: pagination,
       // Wrap in quotes to prevent issue with special characters in frontmatter
       title: `"${event.model.name}"`,
+      sidebar: `\n  order: ${(/[/\\][^/\\]+[/\\]/.test(newURL) ? 0 : 2) + +!isFuncVar}${match ? `\n  label: ${match[3]}` : ""}`
     })
   }
 
