diff --git a/libs/typedoc.ts b/libs/typedoc.ts
index aac6cd1dae63e7b5bc1fda564ac36b4d75e52901..d4f0663747dd69ae007a908b12e8d7598f9abcab 100644
--- a/libs/typedoc.ts
+++ b/libs/typedoc.ts
@@ -161,6 +161,17 @@ function onRendererPageEnd(event: MarkdownPageEvent, outputDirectory: string, pa
     return true
   }
 
+  const old = event.url
+  const match = /([/\\])(?=(.))(?:functions|variables|interfaces|type-aliases|classes|enums)[/\\]([^.]+)\.md$/.exec(old)
+  const newURL = match ? old.slice(0, match.index) + match[1] + match[2] + "-" + match[3] + ".md" : old
+  const isFuncVar = match && (match[2] == "v" || match[2] == "f")
+
+  event.filename = event.filename.slice(0, -old.length) + newURL
+  event.url = newURL
+  event.contents = event.contents
+    ?.replace(/\/(?=(.))(?:functions|variables|interfaces|type-aliases|classes|enums)\//g, "/$1-")
+    .replace(/(?:\.\.\/)+/g, "/")
+
   if (!event.frontmatter) {
     event.contents = addFrontmatter(
       event.contents,
@@ -170,6 +181,7 @@ function onRendererPageEnd(event: MarkdownPageEvent, outputDirectory: string, pa
         prev: pagination,
         // Wrap in quotes to prevent issue with special characters in frontmatter
         title: `"${event.model.name}"`,
+        sidebar: `\n  order: ${(/[/\\][^/\\]+[/\\]/.test(newURL) ? 0 : 2) + +!isFuncVar}${match ? `\n  label: ${match[3]}` : ""}`,
       }),
     )
   }
